### ISSUE TITLE
Terrain elevation and dyanmic line of sight

### ISSUE THREAD

**Issue by royalj** [cyruzzo/AboveVTT#2238]
Not an issue, just a feature request

AboveVTT does well in a lot of areas, especially for open source and free browser based VTT. One thing that is lacking, and is pretty painful to work around is accurately representing line of sight with varying map and token elevation.

E.g. Given the scenario:
* Token A is at elevation 0 and is standing 10 feet from a wall
* The wall has base = 0, top = 10
* Token B is on the other side of the wall, but flying 50 feet in the air
* Token A cannot see Token B

It seems that the software perceive walls as hard obstacles to sight line and does not take into account elevation. To expand on this it would be nice to be able to define an elevation setting for the map itself (like a topographical outline almost), that way as tokens move across the map, their elevation changes to that of where they are standing (could still be overridden to represent flying, burrowing, climbing, etc.)

Not sure if this would be pushing the limits of what AboveVTT could accomplish in the browser or not, but wanted to post this because so far it has been a very enjoyable tool.

**Comment by Azmoria** [comment]
Yeah wall heights just came out and there's certainly improvements to be made. 

This sorta gets into the realm of calculating 3d vision, as you might be able to see someone standing at the edge of a cliff but not 20 ft back from the edge.

 You'd have to calculate the angle and see if it's a greater angle then to the top of the wall. I'll think about it, I'll have to consider performance issues but it might be able to be alleviated by only checking tokens with elevation set. 

This would need a separate vision check/system then we have coded atm. Currently we check against a 2d canvas that has black or white drawn in areas they can/can't see respectively.

It'll probably be low priority and a larger addition so will be awhile before I get around to it. 

For now it might be easiest to set the token to reveal in fog (and make sure it's not set to below darkness) when you want to reveal a token flying over top of things the players can't see past.
